PyGamer_Thermal_Camera/code.py

```diff
@@ -0,0 +1,431 @@
+
+    time.sleep(0.8)  # Show SET status text before setting parameters
+    status_label.text  = ""  # Clear status text
+
+    param_index = 0  # Reset index of parameter to set
+
+    # Select parameter to set
+    while not panel.button.start:
+        while (not panel.button.a) and (not panel.button.start):
+            left, right, up, down = move_buttons(joystick=panel.has_joystick)
+            if up:
+                param_index = param_index - 1
+            if down:
+                param_index = param_index + 1
+            if param_index > 2:
+                param_index = 2
+            if param_index < 0:
+                param_index = 0
+            status_label.text = param_list[param_index][0]
+            image_group[param_index + 66].color = BLACK
+            status_label.color = BLACK
+            time.sleep(0.2)
+            image_group[param_index + 66].color = param_list[param_index][1]
+            status_label.color = WHITE
+            time.sleep(0.2)
+
+        if panel.button.a:  # Button A pressed
+            panel.play_tone(1319, 0.030)  # E6
+        while panel.button.a:  # wait for button release
+            pass
+
+        # Adjust parameter value
+        param_value = int(image_group[param_index + 70].text)
+        while (not panel.button.a) and (not panel.button.start):
+            left, right, up, down = move_buttons(joystick=panel.has_joystick)
+            if up:
+                param_value = param_value + 1
+            if down:
+                param_value = param_value - 1
+            if param_value  > MAX_SENSOR_F:
+                param_value = MAX_SENSOR_F
+            if param_value  < MIN_SENSOR_F:
+                param_value = MIN_SENSOR_F
+            image_group[param_index + 70].text = str(param_value)
+            image_group[param_index + 70].color = BLACK
+            status_label.color = BLACK
+            time.sleep(0.05)
+            image_group[param_index + 70].color = param_list[param_index][1]
+            status_label.color = WHITE
+            time.sleep(0.2)
+
+        if panel.button.a:  # Button A pressed
+            panel.play_tone(1319, 0.030)  # E6
+        while panel.button.a:  # wait for button release
+            pass
+
+    # Exit setup process
+    if panel.button.start:  # Start button pressed
+        panel.play_tone(784, 0.030)  # G5
+    while panel.button.start:  # wait for button release
+        pass
+
+    status_label.text = "RESUME"
+    time.sleep(0.5)
+    status_label.text = ""
+
+    # Display average label and value
+    ave_label.color = YELLOW
+    ave_value.color = YELLOW
+    return int(alarm_value.text), int(max_value.text), int(min_value.text)
+
+def move_buttons(joystick=False):  # Read position buttons and joystick
+    move_r = move_l = False
+    move_u = move_d = False
+    if joystick:  # For PyGamer: interpret joystick as buttons
+        if   panel.joystick[0] > 44000:
```

Comment:
did ya look at cursorcontrol? https://github.com/adafruit/Adafruit_CircuitPython_CursorControl/tree/master/adafruit_cursorcontrol

Reply:
Yes, but it was more complex than the simple up-down setup buttons that were needed for this project.